client/lxd_storage_volumes.go

```diff
@@ -53,8 +54,24 @@ func (r *ProtocolLXD) GetStoragePoolVolumeNamesAllProjects(pool string) ([]strin
 		return nil, err
 	}
 
-	// Parse it.
-	return urlsToResourceNames(u.String(), urls...)
+	names := make(map[string][]string)
+	for _, urlString := range urls {
+		resourceURL, err := url.Parse(urlString)
+		if err != nil {
+			return nil, fmt.Errorf("Could not parse unexpected URL %q: %w", urlString, err)
+		}
+
+		project := resourceURL.Query().Get("project")
+		if project == "" {
+			project = "default"
+		}
+
+		resource := strings.TrimPrefix(resourceURL.Path, fmt.Sprintf("/1.0%s/", u.URL.Path))
```

Comment:
I think it would be worth using `strings.Cut(resourceURL.Path, u.URL.Path)` so that you can detect if the resource.URL doesn't contain the prefix and fail if an unrecognised resource type URL is returned (so that we can detect any regressions in the future).

We could also use strings.Cut in urlsToResourceNames for simplicity.